rebuild/src/addon.h

```diff
@@ -0,0 +1,177 @@
+#ifndef BLST_TS_ADDON_H__
+#define BLST_TS_ADDON_H__
+
+#include <openssl/rand.h>
+
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <string_view>
+
+#include "blst.hpp"
+#include "napi.h"
+
+// TODO: these should come out post PR review
+using std::cout;
+using std::endl;
+
+#define BLST_TS_RANDOM_BYTES_LENGTH 8U
+
+#define BLST_TS_FUNCTION_PREAMBLE(info, env, module)                           \
+    Napi::Env env = info.Env();                                                \
+    Napi::EscapableHandleScope scope(env);                                     \
+    BlstTsAddon *module = env.GetInstanceData<BlstTsAddon>();
+
+#define BLST_TS_IS_INFINITY                                                \
+    Napi::Env env = info.Env();                                                \
+    Napi::EscapableHandleScope scope(env);                                     \
+    return scope.Escape(Napi::Boolean::New(env, _point->IsInfinite()));
```

Comment:
can `_point` be null or invalid? Should this be asserted?

Reply:
No I don't believe so.  In the constructors the only code in there, other than the setting of the unique_ptr is a check to see if a `Napi::External` value was passed to the constructor function (ie it was called from native code because externals are not exposed to the JS side of things.  they can ONLY originate in native functions).  So unless the native constructor initializer list fails (which will throw a hard native error that crashes the process). that is VERY bad btw and should never happen.  it cannot be caught with try/catch on the JS side.... it just stops the world and doesn't do proper cleanup potentially.  Or the isExternal check fails (will throw a JS exception resulting in the instance never getting returned from the constructor and the function context being popped off the stack meaning the value can never be referenced) which means the methods call to IsInfinity() is impossible.  I believe this should be safe unless the `reset()` of the unique_ptr fails but that is marked `noexcept`

Reply:
I am leaving this one unresolved so you can easily find it @LesnyRumcajs .  Please double check that that makes sense.  If not I will add some sort of assert.  Just not sure what kind of error I would throw there though but can figure it out!!

Reply:
As you probably noticed, there are a lot of assumptions. And while these may hold now, the next person modifying this code may completely invalidate them. I'd recommend playing defensively. For example, could `_point` be a `const` pointer, only initialized once in the init list? This way, nobody can reset it later on (at least not so easily). What do you think?

Reply:
Its only a couple of lines so i just moved them into the function body and deleted the macro.  Safer seems better.  Updated in e5010f83d633dd2326d09c40ab98d929a55652cc